### ISSUE TITLE
First video skipped from XBMC with ERROR

### ISSUE THREAD

**Issue by Olivier49** [vitorhirota/plugin.video.globotv#1]
Hi Vitor,

First of all, thanks a lot for this very useful plugin.
I subscribed to Globo Midia, so, I am able to see all videos.
But when I try Novelas-Salve Jorge- Capítulo de terça-feira, dia 08/01/2013, na íntegra, the first video of the four ones is skipped with an error message.
Please find the log there: http://pastebin.com/U5cFdccb

From what I understood the 4 video_ids are:
2334155
2334161
2334175
2334188

But I don't know why, the plugin tries to get video_id=2334198 at first.
Sometimes it can skip this video and go to the next one: 2334155, but most of the time it goes to the second video_id=2334161.

Thanks a lot.
I may help to improve this pluggin, but I'm new to Python and XBMC.

EDIT: the error comes from function "def get_list_items(data):"

```
    try:
        title, date, thumb, duration, descr = eval(cache.get(item['id']))
    except:
        title, date, thumb, duration, descr = [data['title'], '', '', '', '']
```

regards,

Olivier

**Comment by vitorhirota** [comment]
Hey Olivier, 

Glad it's been useful :)

And yeah, I know about this cases and have no idea why it happens.
I did post a note about this when I've launched the previous version on the xbmc forum.
http://forum.xbmc.org/showthread.php?tid=145951&pid=1261422#pid1261422

While watching, I only could go back to the first video using the oficial xbmc remote for android, which lists the items in the playlist, and let me pinpoint which one I would like to play.

The error you point happens when you first visit/play an item, and there's no info of it in the cache yet. So that's why there's a try/except block.

The problem lies here:

```
21:38:35 T:140362281760640   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.globotv/?action=play&video_id=2334198]
```

For some reason, XBMC is classifying as unplayable. Not sure if it's a timeout or something. But when you do go back for that item 0, it plays just fine.

Anyhow, long story short, I'm unable to fix this as of now. But I'll definitely keep looking for a solution.

Cheers.

**Comment by Olivier49** [comment]
Hey Vitor,

Thanks for you answer.
I'm replying here because I still have this error, and you're right, globo didn't change their code. Here's another log: http://pastebin.com/download.php?i=1KZS240P

Sometimes it works after the first error, we can see the first part of the novela, but I'm surprised to see that, instead of the 720 HD stream, we get the iphone 360 one.

For example, for "Salve Jorge - Capítulo de quarta-feira, dia 06/02/2013, na íntegra", it's divided in 3 parts:
2390961
2390971
2391002

If I go to: http://api.globovideos.com/videos/2390961/playlist
I can see all streams, including the 720 HD one: http://flashvideo.globo.com/h264/entretenimentohd/1/salve_jorge/2013/02/06/2390961-web720.mp4

but instead, your plugin takes this one:

http://iphonevideo.globo.com/mp4/entretenimento/4/salve_jorge/2013/02/06/2390961-iphone360.mp4

I know the iphone360 does not need any token. Is it why your plugin picks up this stream ?

I would really like to download myself the videos, as a workaround.

I know I need to go there: http://security.video.globo.com/videos/2390961/hash?resource_id=5112ec74dd2381558a00028c

it gives a token, but then I am unable to calculate the final token, the one that goes after the http://flashvideo.globo.com/h264/entretenimentohd/1/salve_jorge/2013/02/06/2390961-web720.mp4? url.

I looked at your code, but I am new to Python and can't get it...I also looked at the globotv.js I found, but it's very messy...

I've been searching if Globo Midia had a public API website, to explain how things work, but I did not find anything. How did you understand how things work ? alone, with tools ? or is there an API somewhere ? thanks.

I am quite frustrated because I'm a subscriber (Assinante) and I don't feel really free, as I would like to download the videos I paid for :)

I could download some videos with Firefox's "Download Flash & Video" addon, but most of the novelas show a 403 Forbidden, even "Salve Jorge" since 06/02/2013, now...

Thanks and sorry if I speak too much ;)

**Comment by vitorhirota** [comment]
```
Sometimes it works after the first error, we can see the first part of the novela, but I'm surprised to see that, instead of the 720 HD stream, we get the iphone 360 one.
```

I've noticed there's a problem the way I handle the authorization cookies. That's why, even "logged" the 720 don't go through.

I'm currently making major changes, moving the plugin to use xbmcswift2.

It will probably fix these non-HD cases, since I've revised the way to handle authorization, and I also have a clue on what's skipping the first item.

I'll let you know in this thread when it's done.

```
I've been searching if Globo Midia had a public API website, to explain how things work, but I did not find anything. How did you understand how things work ? alone, with tools ? or is there an API somewhere ? thanks.
```

They don't publish any details on their api.
You would have to fiddle around what's going on on their website.
Use JavaScript consoles and debuggers. Chrome and Firefox have great tools for that.

**Comment by Olivier49** [comment]
Hi Vitor,

Thanks a lot ;) that's great news !

**Comment by vitorhirota** [comment]
Fixed on cec24b000a8ccbcb559f40da4b8e6870e3bc6a62

This behaviour was due to the fact that, while creating the playlist, the item being handled was considered to be the first item in the playlist. Since it was never resolved to an actual url, XBMC would skip it to the next item in the playlist.

This was fixed by resolving the url for the first item during the first call that generates the playlist.
